presto-main/src/main/java/com/facebook/presto/util/SpatialJoinUtils.java

```diff
@@ -50,4 +57,37 @@ private static boolean isSupportedSpatialFunction(FunctionCall functionCall)
         String functionName = functionCall.getName().toString();
         return functionName.equalsIgnoreCase(ST_CONTAINS) || functionName.equalsIgnoreCase(ST_INTERSECTS);
     }
+
+    public static boolean spatialFunctionArgumentsMatchJoinNodes(FunctionCall functionCall, PlanNode leftNode, PlanNode rightNode)
+    {
+        List<Expression> arguments = functionCall.getArguments();
+        Verify.verify(arguments.size() == 2);
+
+        Expression firstArgument = arguments.get(0);
+        Expression secondArgument = arguments.get(1);
+        if (!(firstArgument instanceof SymbolReference) || !(secondArgument instanceof SymbolReference)) {
```

Comment:
I don't understand this comment. The previous one says "you cannot expect than any function given to this method will have 2 parameters", then this one says "it would be a bug if argument would be something different than symbol". 

I expect this function to be called only for functions which passed isSupportedSpatialFunction test, hence, they must have exactly 2 arguments. However, the arguments don't necessarily have to be symbol references. They can be constant expressions, or function calls themselves. For example, ST_Contains(ST_GeometryFromText(...), ST_Point(...)).

Reply:
>  expect this function to be called only for functions which passed isSupportedSpatialFunction test

I missed that. Then please use `checkArgument` to ensure that it is truth.

Also I thought that something like:
`For example, ST_Contains(ST_GeometryFromText(...), ST_Point(...)).`  I expected that function calls like `ST_GeometryFromText(...)` and  `ST_Point(...)` are evaluated in project node below this join. If it would be truth then you would need to write this optimizer in the first place. Anyway, I am surprised it didn't work that way.

Reply:
I'll use checkArgument for arguments.size() == 2. Explain query doesn't show ST_GeometryFromText and ST_Point function calls being pushed down in the current version of Presto and I didn't see any rule which might do that.